python/adbc_driver_manager/adbc_driver_manager/dbapi.py

```diff
@@ -914,12 +942,17 @@ def adbc_ingest(
             self._stmt.bind(data)
         elif hasattr(data, "__arrow_c_stream__"):
             self._stmt.bind_stream(data)
-        elif isinstance(data, pyarrow.RecordBatch):
+        elif _lib.is_pycapsule(data, b"arrow_array"):
+            self._stmt.bind(data)
+        elif _lib.is_pycapsule(data, b"arrow_array_stream"):
+            self._stmt.bind_stream(data)
+        elif _has_pyarrow and isinstance(data, pyarrow.RecordBatch):
             array = _lib.ArrowArrayHandle()
             schema = _lib.ArrowSchemaHandle()
             data._export_to_c(array.address, schema.address)
             self._stmt.bind(array, schema)
```

Comment:
should we use `__arrow_c_array__`  and just universally use pycapsules?

Reply:
I bumped up the minimum PyArrow earlier and it seems that always supports this now

Reply:
I'll remove this